planner/core/task.go

```diff
@@ -126,46 +138,139 @@ func (p *PhysicalApply) attach2Task(tasks ...task) task {
 	rTask := finishCopTask(p.ctx, tasks[1].copy())
 	p.SetChildren(lTask.plan(), rTask.plan())
 	p.schema = buildPhysicalJoinSchema(p.JoinType, p)
+	cpuCost := lTask.cost()
+	lCount := lTask.count()
+	if len(p.LeftConditions) > 0 {
+		cpuCost += lCount * cpuFactor
+		lCount *= selectionFactor
+	}
+	rCount := rTask.count()
+	if len(p.RightConditions) > 0 {
+		cpuCost += lCount * rCount * cpuFactor
+		rCount *= selectionFactor
```

Comment:
maybe `rCount` is incorrect when we can use index scan on the inner side table, in which condition the scan range is decided by the correlated outer side join key.

Reply:
But we cannot know the selectivity of the outer key until execution.